cardano-cli/src/Cardano/CLI/Run/Key.hs

```diff
@@ -205,6 +207,19 @@ runNonExtendedKeyCmd
         writeLazyByteStringFile vkf' $
           textEnvelopeToJSON descr vk
 
+runGenerateMnemonicCmd :: Cmd.KeyGenerateMnemonicCmdArgs -> ExceptT KeyCmdError IO ()
```

Comment:
I think when writing to file we should at least set restrictive permissions. Can you get in touch with the wallet team and ask them what precautions they took when generating mnemonics?